### DISCUSSION TITLE
Nix package is signifficantly slower than if I compile it myself inside of a container

### DISCUSSION THREAD

**Discussion by nonetrix** [fastfetch-cli/fastfetch#797] [Q&A] [answered]
![image](https://github.com/fastfetch-cli/fastfetch/assets/45698918/d2833167-cd0b-45cf-ad78-7835dcc427ce)
I imagine this is difference in what cflags are being used, but it seems to make a huge difference. I assume there is a way to overlay the package and get the same result, but I haven't been able to figure out how to do this. The difference is night and day though in person and it's rather annoying, it seems to have a really hard time at about the half way mark, something isn't being optimized with the compiler clearly

And here is the identical logo to be safe, in this case if anything the version from source should be slower since it's printing more lines
![image](https://github.com/fastfetch-cli/fastfetch/assets/45698918/de189d92-c109-4499-9942-5a07902b4022)

**Comment by CarterLi** [comment]
`--stat` will print time used for each modules.

**Comment by CarterLi** [comment]
fastfetch uses shell script to detect the number of nix packages.

https://github.com/fastfetch-cli/fastfetch/blob/dev/src/detection/packages/packages_linux.c#L131-L133

It's expected to be slow. [The code was contributed by nix users](https://github.com/fastfetch-cli/fastfetch/issues/195) but personally I don't use nix. Any contribution is welcome.

**Reply by nonetrix** [reply]
Yep seems to be that :/

**Comment by nonetrix** [comment, marked as answer]
https://github.com/fastfetch-cli/fastfetch/pull/817
